### PR TITLE
Switch MATLAB tree-sitter parser

### PR DESCRIPTION
**Pull request by acristoffers**
Product of #7386.

I believe I got the highlight right, or at least now I get a nice rainbow-vomit when using the `doom_acario_dark` theme. And no more quirks.

Textobjects and indents were tested and are inline with what I found in other grammars. That is to say "could you please take a look at the indents because I think it is broken, but it is defined the same as other languages so I don't know if it's broken or if it's just me who doesn't know how this editor works".

 I simply copied the other queries and did not adapt them or anything. I not even know if helix supports them.

image

### CODE REVIEW

File: runtime/queries/matlab/highlights.scm [hunk 1]
```diff
@@ -1,97 +1,127 @@
...
+
+; Punctuation
+
+[ ";" "," "." ] @punctuation.delimiter
+[ "(" ")" "[" "]" "{" "}" ] @punctuation.bracket
+
+; Literals
+
+(escape_sequence) @escape
+(formatting_sequence) @escape
```
Comment (the-mikedavis) [inline runtime/queries/matlab/highlights.scm:106]:
```suggestion
(escape_sequence) @constant.character.escape
(formatting_sequence) @constant.character.escape
```

File: runtime/queries/matlab/highlights.scm [hunk 2]
```diff
@@ -1,97 +1,127 @@
...
 (string) @string
+(number) @constant.numeric.float
+(boolean) @constant.builtin.boolean
 
-;; Comment
-(comment) @comment
+; Comments
+
+[ (comment) (line_continuation) ] @comment.line @spell
```
Comment (the-mikedavis) [inline runtime/queries/matlab/highlights.scm:113]: The `@spell` capture should probably be removed. I think if we do spell-checking with tree-sitter in core then we'll use a separate query file for the spelling queries than the highlight queries.

File: runtime/queries/matlab/injections.scm [hunk 3]
```diff
@@ -0,0 +1 @@
+(comment) @comment
```
Comment (the-mikedavis) [inline runtime/queries/matlab/injections.scm:1]:
```suggestion
((comment) @injection.content
 (#set! injection.language "comment"))
```

We follow tree-sitter's injection format: https://tree-sitter.github.io/tree-sitter/syntax-highlighting#language-injection. I believe this is neovim's style. (I'm not sure if emacs follows nvim here.)

File: runtime/queries/matlab/locals.scm
Comment (the-mikedavis) [inline runtime/queries/matlab/locals.scm:1]: For locals we also follow upstream tree-sitter: https://tree-sitter.github.io/tree-sitter/syntax-highlighting#local-variables

So there should only be `@local.reference`, `@local.definition` and `@local.scope`. We don't differentiate between variable/function/parameter definitions and references.

**Comment by the-mikedavis** [comment]
If you run `cargo xtask docgen` and commit the results that will clear up the X in the docs CI

File: runtime/queries/matlab/highlights.scm (hunk 1 above)
Comment (acristoffers) [reply runtime/queries/matlab/highlights.scm:106]: I skipped them and forgot to go back 🤦‍♂️

File: runtime/queries/matlab/highlights.scm (hunk 2 above)
Comment (acristoffers) [reply runtime/queries/matlab/highlights.scm:113]: I saw it was not in the list, but I wasn't sure whether it was used or not. I'll remove it.

File: runtime/queries/matlab/locals.scm
Comment (acristoffers) [reply runtime/queries/matlab/locals.scm:1]: I forgot I had to change those for neovim. I may have the original in an old commit, I'll look for it.

File: runtime/queries/matlab/locals.scm [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  (function_arguments
+    (identifier)* @local.definition
+    ("," (identifier) @local.definition)*)?) @local.scope
+
+(assignment variable: ((function_call
+                         name: (identifier) @local.definition)))
+(assignment variable: ((struct . [(function_call
+                                    name: (identifier) @local.definition)
+                                  (identifier) @local.definition])))
+(assignment variable: (_) @local.definition)
```
[the remark this answers is not in the document]
Comment (acristoffers) [reply runtime/queries/matlab/locals.scm]: I forgot it got changed to `left`, sorry.

File: runtime/queries/matlab/locals.scm [hunk 5]
```diff
@@ -0,0 +1,19 @@
+(function_definition name: (identifier) @local.definition ?) @local.scope
+(function_arguments (identifier)* @local.definition)
+
+(lambda (arguments (identifier) @local.definition)) @local.scope
+
+(assignment left: ((function_call
+                         name: (identifier) @local.definition)))
+(assignment left: ((struct . [(function_call
+                                    name: (identifier) @local.definition)
+                                  (identifier) @local.definition])))
```
Comment (the-mikedavis) [inline runtime/queries/matlab/locals.scm:10]: It looks like the query analysis is upset about this `struct` node. Was this renamed? I don't see it in the `grammar.js`

Comment (acristoffers) [reply runtime/queries/matlab/locals.scm]: I finally had the decency of copying the files over and running `tree-sitter test` before committing, so there should be no more errors like that. I'm sorry about this, it was a busy day and I'm more tired that I thought.

**Review by the-mikedavis — APPROVED** [review]
Thank you!

**Comment by the-mikedavis** [comment]
It's not necessary: since it's a single commit we will probably squash+merge which will rename the commit and include the PR number

**Review by pascalkuthe — APPROVED** [review]
Thanks!

**Comment by acristoffers** [comment]
One question: do you update the tagged commit automatically or do I have to make pull-requests?

**Comment by pascalkuthe** [comment]
It's done manually since grammar changes usually also require changes to the queries

**Pull request merged**